### ISSUE TITLE
worksheet name with `&`

### ISSUE THREAD

**Issue by JanMarvin** [JanMarvin/openxlsx2#252]
In a sheet with `&` `get_sheet_names()` returns `&amp;`. It's a named character and the names are correct. Maybe we should return only the names.

**Comment by allmedia-nz** [comment]
This is very annoying, particularly if you copy sheet names into an excel cell. It means wb_get_sheet_names(wb) output <> read_xlsx(wb2) when wb2 contains a list of sheet names . This is not a problem with openpyxl

**Comment by JanMarvin** [comment]
@allmedia-nz I marked your comment as off-topic for two reasons. a) It does not provide any new knowledge. Nothing has changed since June 2022 and `wb |> wb_get_sheet_names() |> names()` has always worked. b) if it works with openpyxl, then by all means use it. Nobody in the whole world forces you to use our open source solution. If one solution doesn't work, then maybe another one will. And just complaining ("This is very annoying...") is not helpful in any way.

But hey, I opened the issue, obviously I wasn't happy with it either. That's why I opened #724, which should solve the problem. But the function is used in several places internally, and the unescaped xml special characters caused unpleasant problems along the way, so it will have to wait until the next release, when hopefully we've had a chance for more extensive testing.
